packages/demos/stories/DragDrop.tsx

```diff
@@ -0,0 +1,135 @@
+/**
+ * © 2020 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ClayList from '@clayui/list';
+import {XYCoord} from 'dnd-core';
+import update from 'immutability-helper';
+import React, {useState, useCallback, useRef} from 'react';
+import {useDrag, useDrop, DropTargetMonitor} from 'react-dnd';
+
+export interface IDraggableListItemProps {
+	/**
+	 * The text displayed in the List Item
+	 */
+	text: string;
+	/**
+	 * Index of the List Item in order to keep track of it's positioning within the array
+	 */
+	index: number;
+	/**
+	 * Function that handles the moving of List Items
+	 */
+	moveListItem: (dragIndex: number, hoverIndex: number) => void;
+}
+
+interface IDragItem {
+	index: number;
+	type: string;
+}
+
+const DraggableListItem: React.FC<IDraggableListItemProps> = ({
+	index,
+	moveListItem,
+	text,
+}) => {
+	const ref = useRef<HTMLDivElement>(null);
+
+	const [, drop] = useDrop({
```

Comment:
Lets avoid using empty args, some people have had issues building with an empty arg. You might run into eslint complaining about an unused var, in which case we might want to add this rule https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars.md#args-after-used

Reply:
Hey @kresimir-coko could you look into this as well?